### PR TITLE
consolidate tutorials. give sequence numbers to nameservice pages 

### PR DESCRIPTION
**Pull request by hschoenburg**
Ok, this is what the new vuepress build looks like. 
Screen Shot 2019-09-19 at 1 16 56 PM

This PR is pending a new tutorials website with its own build process. The Tutorials need to come OUT of the SDK docs website so that we can continue to add more. 

**Once merged, this repo should be renamed to `cosmos/sdk-tutorials`**

### CODE REVIEW

**Comment by marbar3778** [comment]
what is blocking this pr? with the new site and docs going up soon, we should get this merged @hschoenburg, @mircea-c i believe we need circleci for this, if you enable on the other pr, i can write out based on the sdk one

**Comment by mircea-c** [comment]
@marbar3778 seems there's a lot of confusion here. Is the new tutorials website with it's own build process supposed to reside in this repo right here? Like I mentioned in the other PR, I don't see any website build target in the Makefile here.

**Comment by fadeev** [comment]
@hschoenburg I don't have the right permissions to push to this repo.

You can do `npm init && npm install --save vuepress-theme-cosmos`

And paste in `.vuepress/config.js`:

```
module.exports = {
  theme: "cosmos",
  markdown: {
    anchor: {
      permalinkSymbol: ""
    }
  },
  themeConfig: {
    autoSidebar: false,
    sidebar: [
      {
        title: "Tutorials",
        children: [
          {
            title: "Hellochain",
            path: "/hellochain/tutorial/"
          },
          {
            title: "Nameservice",
            path: "/nameservice/tutorial/"
          }
        ]
      }
    ]
  }
};
```

That should do it.

**Comment by marbar3778** [comment]
could we merge this for the consolidation factor, it looks good so far and then we can worry about stuff later

**Comment by marbar3778** [comment]
LICENSE → nameservice/LICENSE this should stay at the route, will review it more in-depth in the coming days

**Comment by marbar3778** [comment]
By consolidating it into one repo who will manage updating versions, this will be a large endeavor depending on the amount of breaking changes, If we want to version the tutorials this means on new breaking changes to the SDK we have to update them all at once and cut a release, this may slow down getting a release out

File: utils/starter/app_starter.go [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+	config.SetRoot(DefaultNodeHome)
+
+	server.UpgradeOldPrivValFile(config)
+
+	_, _, err := genutil.InitializeNodeValidatorFiles(config)
+	if err != nil {
+		panic(err)
+	}
+
+	privValidator := pvm.LoadOrGenFilePV(
```
Comment (marbar3778) [inline utils/starter/app_starter.go]: should add godoc that this is injecting a validator

**Comment by marbar3778** [comment]
should we rename the repo before this is merged so the go.mod files can be made with the correct path

**Comment by marbar3778** [comment]
I dont believe this is blocked anymore? @hschoenburg

**Comment by okwme** [comment]
Might still need to update tutorial markdown files...

File: .vuepress/config.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  },
+  themeConfig: {
+    autoSidebar: false,
+    sidebar: [
+      {
+        title: 'Tutorials',
+        children: [
+          {
+            title: 'Hellochain',
+            path: '/hellochain/tutorial/'
```
Comment (fadeev) [inline .vuepress/config.js]:
```suggestion
            path: '/hellochain/tutorial/',
           directory: true,
```

File: .vuepress/config.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+          },
+          {
+            title: 'Nameservice',
+            path: '/nameservice/tutorial/'
```
Comment (fadeev) [inline .vuepress/config.js]:
```suggestion
            path: '/nameservice/tutorial/',
            directory: true
```

File: package.json [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    "golang"
+  ],
+  "author": "All in Bits",
+  "license": "ISC",
+  "bugs": {
+    "url": "https://github.com/cosmos/sdk-application-tutorial/issues"
+  },
+  "homepage": "https://github.com/cosmos/sdk-application-tutorial#readme",
+  "dependencies": {
+    "vuepress-theme-cosmos": "^1.0.29"
```
Comment (fadeev) [inline package.json]:
```suggestion
    "vuepress-theme-cosmos": "^1.0.30"
```

File: nameservice/LICENSE [hunk 5]
```diff
@@ -0,0 +1,204 @@
+Cosmos-SDK
```
Comment (marbar3778) [inline nameservice/LICENSE]: remove

File: hellochain/x/greeter/types/types.go [hunk 6]
```diff
@@ -0,0 +1,63 @@
+package types
+
```
Comment (marbar3778) [inline hellochain/x/greeter/types/types.go]: I think we should still follow the module spec for the module layout

**Comment by marbar3778** [comment]
still need to order the pages

File: hellochain/cmd/hcd/main.go [hunk 7]
```diff
@@ -4,7 +4,7 @@ import (
 	"github.com/tendermint/tendermint/libs/cli"
```
Comment (marbar3778) [inline hellochain/cmd/hcd/main.go]: why was this done? these utils are specific to hellochain and have nothing to do with the nameservice tutorial?
everything tutorial specific goes with the tutorial not at the root

File: hellochain/cmd/hccli/main.go [hunk 8]
```diff
@@ -0,0 +1,30 @@
+package main
+
+import (
+	"github.com/tendermint/tendermint/libs/cli"
+
+	app "github.com/cosmos/sdk-tutorials/hellochain"
+	"github.com/cosmos/sdk-tutorials/hellochain/x/greeter"
+	"github.com/cosmos/sdk-tutorials/utils/starter"
```
Comment (marbar3778) [inline hellochain/cmd/hccli/main.go]: why was this reverted?

File: nameservice/tutorial/00-intro.md [hunk 9]
```diff
@@ -65,28 +69,4 @@ git init
 
 Then, just follow along! The first step describes the design of your application. If you want to jump directly to the coding section, you can start with the [second step](./keeper.md)
 
-### Tutorial parts
-
```
Comment (marbar3778) [inline nameservice/tutorial/00-intro.md]: love this ❤️

File: go.mod [hunk 10]
```diff
@@ -1,18 +1,13 @@
-module github.com/cosmos/sdk-application-tutorial
+module github.com/cosmos/sdk-tutorials
```
Comment (marbar3778) [inline go.mod]: remove this file

**Comment by hschoenburg** [comment]
ok! i think i got to everyones comments. thank you for the reviews and commits!

starter is now inside hellochain where it belongs.

the paths and links should all be working
both applications build

@mircea-c from your comment earlier, does this repo might need a Makefile in its root?

The site  builds with "vuepress build" into .vuepress/dist

What more is needed to get this thing ready for deployment?

File: go.mod (hunk 10 above)
Comment (marbar3778) [reply go.mod]: bump ^^ this file is not needed

**Comment by marbar3778** [comment]
resolve merge conflicts then lets merge and then open a following pr of the circle ci stuff, this PR is huge already

**Pull request merged**